src/cne/elector.rs

```diff
@@ -1,5 +1,27 @@
 use crate::cne::Citizenship;
-use serde::{Serialize, Deserialize};
+use scraper::{Html, Selector, html::Select, element_ref::ElementRef};
+use serde::{Deserialize, Serialize};
+
+fn select_field<'a>(document: &'a mut Select) -> scraper::element_ref::ElementRef<'a> {
+  let selector = Selector::parse("tr td:last-child").unwrap();
+  document.next().unwrap().select(&selector).next().unwrap()
+}
+
+fn select_bold_text<'a>(field: &'a mut ElementRef<'a>) -> scraper::element_ref::ElementRef<'a> {
+  let selector = Selector::parse("b").unwrap();
+  field.select(&selector).next().unwrap()
+}
+
+fn select_font_tag<'a>(field: &'a mut ElementRef<'a>) -> scraper::element_ref::ElementRef<'a> {
+  let selector = Selector::parse("font").unwrap();
+  field.select(&selector).next().unwrap()
+}
+
+fn verify_citizenship_error(html_string: &str) -> bool {
```

Comment:
This should be validated before getting here. Is the request responsibility to check if the response is successful or not, `Elector` should be responsible of handling `Elector` related logic, this validation belongs to the request itself.